src/main/java/org/sonarlint/intellij/analysis/JavaAnalysisConfigurator.java

```diff
@@ -50,118 +47,42 @@
-      if (encoding != null) {
-        return encoding;
-      }
-    }
-    return Charset.defaultCharset();
-  }
-
-  private static List<ClientInputFile> getInputFiles(Project p, ModuleRootManager moduleRootManager, Collection<VirtualFile> filesToAnalyze) {
-    Collection<String> testFolderPrefix = findTestFolderPrefixes(moduleRootManager);
-    List<ClientInputFile> inputFiles = new LinkedList<>();
-
-    for (VirtualFile f : filesToAnalyze) {
-      boolean test = isTestFile(testFolderPrefix, f);
-      Charset charset = getEncoding(p, f);
-      inputFiles.add(new DefaultInputFile(f, test, charset));
-    }
-
-    return inputFiles;
-  }
-
-  private static boolean isTestFile(Collection<String> testFolderPrefix, VirtualFile f) {
-    String filePath = f.getPath();
-    for (String testPrefix : testFolderPrefix) {
-      if (filePath.startsWith(testPrefix)) {
-        return true;
-      }
-    }
-    return false;
-  }
-
-  private static void configureModuleSettings(@NotNull Module ijModule, @NotNull Map<String, String> properties) {
+  public Map<String, String> configure(@NotNull Module ijModule) {
+    Map<String, String> properties = new HashMap<>();
     configureLibraries(ijModule, properties);
     configureBinaries(ijModule, properties);
     configureJavaSourceTarget(ijModule, properties);
+    return properties;
   }
 
   private static void configureJavaSourceTarget(final Module ijModule, Map<String, String> properties) {
     try {
-      final String languageLevel = getLanguageLevelOption(ApplicationManager.getApplication()
-        .runReadAction((Computable<LanguageLevel>) () -> EffectiveLanguageLevelUtil.getEffectiveLanguageLevel(ijModule)));
+      LanguageLevel languageLevel = ApplicationManager.getApplication()
+        .runReadAction((Computable<LanguageLevel>) () -> EffectiveLanguageLevelUtil.getEffectiveLanguageLevel(ijModule));
+      final String languageLevelStr = getLanguageLevelOption(languageLevel);
       String bytecodeTarget = CompilerConfiguration.getInstance(ijModule.getProject()).getBytecodeTargetLevel(ijModule);
       if (StringUtil.isEmpty(bytecodeTarget)) {
         // according to IDEA rule: if not specified explicitly, set target to be the same as source language level
-        bytecodeTarget = languageLevel;
+        bytecodeTarget = languageLevelStr;
       }
-      properties.put(JAVA_SOURCE_PROPERTY, languageLevel);
+      properties.put(JAVA_SOURCE_PROPERTY, languageLevelStr);
       properties.put(JAVA_TARGET_PROPERTY, bytecodeTarget);
-    } catch (NoClassDefFoundError e) {
-      // CompilerConfiguration not available for example in PHP Storm
+    } catch (BootstrapMethodError | NoClassDefFoundError e) {
+      // (DM): some components are not available in some flavours, for example ConpilerConfiguration and Language Level in PHP storm or CLion.
+      // Even though this class should now only be loaded when the Java extensions are available, I leave this to be safe
```

Comment:
Not really, because we know that this section is currently the only one needing this "guard" for it to run in all the flavors. If this class gets loaded in an environment where it should not, this will be enough to avoid runtime errors. I prefer to leave during 1 release until I'm more confident about this change.